java-example/src/main/java/org/onflow/examples/java/AccessAPIConnector.java

```diff
@@ -79,10 +80,28 @@ private FlowTransactionResult waitForSeal(FlowId txID) {
     }
 
     private FlowAddress getAccountCreatedAddress(FlowTransactionResult txResult) {
-        String addressHex = (String) txResult.getEvents().get(0).getEvent().getValue().getFields()[0].getValue().getValue();
-        return new FlowAddress(addressHex.split("\\.")[1]);
+        for (FlowEvent event : txResult.getEvents()) {
+            if ("flow.AccountCreated".equals(event.getType())) {
+                String addressHex = getString(event);
+                return new FlowAddress(addressHex);
+            }
+        }
+        throw new RuntimeException("No flow.AccountCreated event found in the transaction result.");
     }
 
+    @NotNull
+    private static String getString(FlowEvent event) {
```

Comment:
```suggestion
    private static String extractEventAddress(FlowEvent event) {
```

Naming is pretty vague here, proposing something clearer